test/utils/floating_point_utils.hpp

```diff
@@ -0,0 +1,69 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+namespace fizzy::test
+{
+template <typename T, typename = std::enable_if<std::is_floating_point_v<T>>>
+struct FPUtils
+{
+    using Limits = std::numeric_limits<T>;
+
+    static constexpr auto NumMantissaBits = Limits::digits - 1;
+    static constexpr auto MantissaMask = (uint64_t{1} << NumMantissaBits) - 1;
+
+    static constexpr auto CanonicalNaNPayload = uint64_t{1} << (NumMantissaBits - 1);
```

Comment:
Do you have a good link about where it's defined as canonical?

Reply:
Link added.